x/bep3/types/swap_test.go

```diff
@@ -44,95 +44,205 @@ func (suite *AtomicSwapTestSuite) SetupTest() {
 }
 
 func (suite *AtomicSwapTestSuite) TestNewAtomicSwap() {
-	type args struct {
-		amount              sdk.Coins
-		randomNumberHash    tmbytes.HexBytes
-		expireHeight        int64
-		timestamp           int64
-		sender              sdk.AccAddress
-		recipient           sdk.AccAddress
-		recipientOtherChain string
-		senderOtherChain    string
-		closedBlock         int64
-		status              types.SwapStatus
-		crossChain          bool
-		direction           types.SwapDirection
-	}
 	testCases := []struct {
-		description string
-		args        args
-		expectPass  bool
+		msg        string
+		swap       types.AtomicSwap
+		expectPass bool
 	}{
 		{
-			"normal",
-			args{
-				amount:              cs(c("bnb", 50000)),
-				randomNumberHash:    suite.randomNumberHashes[0],
-				expireHeight:        int64(360),
-				timestamp:           suite.timestamps[0],
-				sender:              suite.addrs[0],
-				recipient:           suite.addrs[5],
-				recipientOtherChain: "bnb1urfermcg92dwq36572cx4xg84wpk3lfpksr5g7",
-				senderOtherChain:    "bnb1uky3me9ggqypmrsvxk7ur6hqkzq7zmv4ed4ng7",
-				closedBlock:         0,
-				status:              types.Open,
-				crossChain:          true,
-				direction:           types.Incoming,
+			"valid Swap",
+			types.AtomicSwap{
+				Amount:              cs(c("bnb", 50000)),
+				RandomNumberHash:    suite.randomNumberHashes[0],
+				ExpireHeight:        360,
+				Timestamp:           suite.timestamps[0],
+				Sender:              suite.addrs[0],
+				Recipient:           suite.addrs[5],
+				RecipientOtherChain: "bnb1urfermcg92dwq36572cx4xg84wpk3lfpksr5g7",
+				SenderOtherChain:    "bnb1uky3me9ggqypmrsvxk7ur6hqkzq7zmv4ed4ng7",
+				ClosedBlock:         1,
+				Status:              types.Open,
+				CrossChain:          true,
+				Direction:           types.Incoming,
 			},
 			true,
 		},
+		{
+			"invalid amount",
+			types.AtomicSwap{
+				Amount: sdk.Coins{sdk.Coin{Denom: "BNB", Amount: sdk.NewInt(10)}},
+			},
+			false,
+		},
```

Comment:
Fails even with a valid amount due to `the length of random number hash should be 32`. Recommend checking that the returned error matches expected error.

Reply:
I double-checked and the error matches the message